src/structures/gdt.rs

```diff
@@ -315,8 +330,12 @@ impl Descriptor {
         // base
         low.set_bits(16..40, ptr.get_bits(0..24));
         low.set_bits(56..64, ptr.get_bits(24..32));
-        // limit (the `-1` in needed since the bound is inclusive)
-        low.set_bits(0..16, (size_of::<TaskStateSegment>() - 1) as u64);
+        // limit (the `-1` is needed since the bound is inclusive)
+        let iomap_limit = tss.iomap_base as u64 + iomap_size as u64 - 1;
+        low.set_bits(
+            0..16,
+            cmp::max(mem::size_of::<TaskStateSegment>() as u64, iomap_limit),
```

Comment:
We need to do the subtraction after the ma

Reply:
Right, good catch!